### PR TITLE
chore: lint tests in modules other than gamm

### LINKED ISSUE #5061: lint tests

### PR DESCRIPTION
**Pull request by faddat**
## What is the purpose of the change

Works toward closing, but does not close #5061 

This PR is to keep momentum going without blocking on some type assertions in x/gamm.  It lints all modules other 
than x/gamm.  Will continue on with x/gamm in #5127.

## Testing and Verifying

If this passes the test suite, then it works because it focuses on linting tests. 

## Documentation and Release Note

No changelog because there's no change to the application. 

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

**Comment by faddat** [comment]
@mattverse -- fyi, I've tried "jiggling" the changelog reminder but to no avail.  I think we are g2g here, and that #5127 can be looked at again after #5140.  


Oh noes the tests just failed :/ .  Moment...

**Comment by faddat** [comment]
Oh noes the tests just failed :/ .  Moment...

**Comment by faddat** [comment]
@czarcas7ic should I use a different label or add a changelog? 

<3

**Comment by czarcas7ic** [comment]
@faddat  yeah sorry about that, we switched to a no-changelog that with the A prefix. Just added it now.

**Comment by faddat** [comment]
hey no problem @czarcas7ic <3

File: x/concentrated-liquidity/swaps.go [hunk 1]
```diff
@@ -254,15 +259,6 @@ func (k Keeper) computeOutAmtGivenIn(
...
 	}
-
-	hasPositionInPool, err := k.HasAnyPositionForPool(ctx, poolId)
-	if err != nil {
-		return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, err
-	}
-	if !hasPositionInPool {
-		return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, types.NoSpotPriceWhenNoLiquidityError{PoolId: poolId}
-	}
-
```
Comment (pysel) [inline x/concentrated-liquidity/swaps.go:265]: why removing this?

File: x/concentrated-liquidity/swaps.go [hunk 2]
```diff
@@ -416,31 +412,23 @@ func (k Keeper) computeOutAmtGivenIn(
...
 
-// computeInAmtGivenOut calculates tokens to be swapped in given the desired token out and fee deducted. It also returns
+// calcInAmtGivenOut calculates tokens to be swapped in given the desired token out and fee deducted. It also returns
 // what the updated tick, liquidity, and currentSqrtPrice for the pool would be after this swap.
-// Note this method is mutative, some of the tick and accumulator updates get written to store.
-// However, there are no token transfers or pool updates done in this method. These mutations are performed in swapOutAmtGivenIn.
-func (k Keeper) computeInAmtGivenOut(
+// Note this method is non-mutative, so the values returned by calcInAmtGivenOut do not get stored
+// Instead, we return writeCtx function so that the caller of this method can decide to write the cached ctx to store or not.
+func (k Keeper) calcInAmtGivenOut(
```
Comment (pysel) [inline x/concentrated-liquidity/swaps.go:419]: I am not sure why we are refactoring this function?

File: x/concentrated-liquidity/swaps_test.go [hunk 3]
```diff
@@ -1602,38 +1602,6 @@ func (s *KeeperTestSuite) TestComputeAndSwapOutAmtGivenIn() {
...
-	_, _, _, _, _, err := s.App.ConcentratedLiquidityKeeper.SwapInAmtGivenOut(
-		s.Ctx, s.TestAccs[0], pool,
-		DefaultCoin0, DefaultCoin1.Denom,
-		sdk.ZeroDec(), sdk.ZeroDec(),
-	)
-
-	s.Require().Error(err)
-	s.Require().ErrorIs(err, types.NoSpotPriceWhenNoLiquidityError{PoolId: pool.GetId()})
-}
-
```
Comment (pysel) [inline x/concentrated-liquidity/swaps_test.go:1636]: why removing tests?

File: x/concentrated-liquidity/swaps_test.go [hunk 4]
```diff
@@ -2515,70 +2483,18 @@ func (s *KeeperTestSuite) TestCalcInAmtGivenOut_NonMutative() {
...
-					osmomath.BigDecFromSDKDec(feeAccum.GetValue().AmountOf(test.tokenIn.Denom)),
+					osmomath.BigDecFromSDKDec(feeAccum.GetValue().AmountOf(test.tokenInDenom)),
 				),
 			)
-		})
-	}
-}
-
-// TestComputeInAmtGivenOut tests that ComputeInAmtGivenOut successfully performs state changes as expected.
-func (s *KeeperTestSuite) TestComputeInAmtGivenOut() {
```
Comment (pysel) [inline x/concentrated-liquidity/swaps_test.go:2526]: why removing this test? I think we should also test outGivenIn logic

Comment (pysel) [reply x/concentrated-liquidity/swaps_test.go:2526]: https://github.com/osmosis-labs/osmosis/pull/5207/files#diff-1a552fcc0a3d36556402345e840b658a7daccb101d7f73bca9ee2e3057ade2e6R2427 we have `inGivenOut` test here

File: x/concentrated-liquidity/incentives_test.go [hunk 5]
```diff
@@ -3738,7 +3738,7 @@ func (s *KeeperTestSuite) TestGetAllIncentiveRecordsForUptime() {
 				curUptimeRecords, err := clKeeper.GetAllIncentiveRecordsForUptime(s.Ctx, poolId, supportedUptime)
 				s.Require().NoError(err)
 
-				retrievedRecordsByUptime = append(retrievedRecordsByUptime, curUptimeRecords...)
+				retrievedRecordsByUptime = append(retrievedRecordsByUptime, curUptimeRecords...) //nolint:makezero,staticcheck
```
Comment (mattverse) [inline x/concentrated-liquidity/incentives_test.go:3741]: I remember asking this somewhere else, but dont remember if I ever got to the answer, what does makezero do here?

File: x/concentrated-liquidity/swaps.go [hunk 6]
```diff
@@ -187,7 +187,7 @@ func (k *Keeper) swapInAmtGivenOut(
 	swapFee sdk.Dec,
 	priceLimit sdk.Dec,
 ) (calcTokenIn, calcTokenOut sdk.Coin, currentTick int64, liquidity, sqrtPrice sdk.Dec, err error) {
-	tokenIn, tokenOut, newCurrentTick, newLiquidity, newSqrtPrice, err := k.computeInAmtGivenOut(ctx, desiredTokenOut, tokenInDenom, swapFee, priceLimit, pool.GetId())
+	writeCtx, tokenIn, tokenOut, newCurrentTick, newLiquidity, newSqrtPrice, err := k.calcInAmtGivenOut(ctx, desiredTokenOut, tokenInDenom, swapFee, priceLimit, pool.GetId())
```
Comment (mattverse) [inline x/concentrated-liquidity/swaps.go:190]: we don't need to call CalcInAmtGivenOut and then call writeCtx, instead, here we pass in computeInAmtGivenOut directlyt with the context we are using, thus mutating state anyways

File: x/concentrated-liquidity/swaps.go [hunk 7]
```diff
@@ -197,6 +197,11 @@ func (k *Keeper) swapInAmtGivenOut(
 		return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, types.InvalidAmountCalculatedError{Amount: tokenIn.Amount}
 	}
 
+	// N.B. making the call below ensures that any mutations done inside calcInAmtGivenOut
+	// are written to store. If this call were skipped, calcInAmtGivenOut would be non-mutative.
+	// An example of a store write done in calcInAmtGivenOut is updating ticks as we cross them.
+	writeCtx()
+
```
Comment (mattverse) [inline x/concentrated-liquidity/swaps.go:204]: ditto for here

File: x/concentrated-liquidity/swaps.go [hunk 8]
```diff
@@ -229,7 +234,7 @@ func (k Keeper) CalcInAmtGivenOut(
 	swapFee sdk.Dec,
 ) (tokenIn sdk.Coin, err error) {
 	cacheCtx, _ := ctx.CacheContext()
-	tokenIn, _, _, _, _, err = k.computeInAmtGivenOut(cacheCtx, tokenOut, tokenInDenom, swapFee, sdk.ZeroDec(), poolI.GetId())
+	_, tokenIn, _, _, _, _, err = k.calcInAmtGivenOut(cacheCtx, tokenOut, tokenInDenom, swapFee, sdk.ZeroDec(), poolI.GetId())
```
Comment (mattverse) [inline x/concentrated-liquidity/swaps.go:237]: ditto for here

**Review by mattverse — COMMENTED** [review]
The lints lgtm, but I dont think we should be changing the swap functions for CL.(or at least if there are notable changes, let's get them in a separate PR please!)

File: x/concentrated-liquidity/swaps.go (hunk 8 above)
Comment (faddat) [reply x/concentrated-liquidity/swaps.go:237]: came in a merge from main, I think.  

Let me just re-do cliq tho.

BTW I'd love to rename concentrate-liqudity to cliq

it's so fire

File: x/concentrated-liquidity/swaps_test.go (hunk 3 above)
Comment (faddat) [reply x/concentrated-liquidity/swaps_test.go:1636]: becasue you're a good reviewer (I will restore now, this is def an error, thank you)

**Comment by faddat** [comment]
fyi to @mattverse & @pysel thanks to both of you for your reviews.  

Since issues centered around cliq, I've checked that module out of main and am relinting.

**Comment by faddat** [comment]
fyi to @mattverse & @pysel thanks to both of you for your reviews.  

Since issues centered around cliq, I've checked that module out of main and am relinting.

**Comment by faddat** [comment]
Note: I really need to figure out what I am doing that breaks this... tomorrow

**Comment by faddat** [comment]
Closing in favor of #5127

**Pull request closed without merging**